### PR TITLE
[Outlook] (notifications) Fix insight notification snippet

### PR DESCRIPTION
**Pull request by samantharamon**
Fixes the snippet that adds an insight notification to a mail item. The snippet now references command IDs available in the Script Lab manifest. This PR also cleans up the task pane UI to provide users with instructions on how to effectively run the snippet.

### CODE REVIEW

File: samples/outlook/35-notifications/add-getall-remove.yaml [hunk 1]
```diff
@@ -127,34 +148,51 @@ template:
         </section>
 
         <section class="samples ms-font-m">
-          <h3>Try it out</h3>
+          <h2>Try it out</h3>
```
Comment (kwame-ansong) [inline samples/outlook/35-notifications/add-getall-remove.yaml:151]: From my view, it appears we have an <h2> tag being closed with an </h3> tag. Was the intent to close it with an </h2> tag instead?

File: snippet-extractor-output/snippets.yaml [hunk 2]
```diff
@@ -10456,20 +10456,44 @@
...
         type: Office.MailboxEnums.ItemNotificationMessageType.InsightMessage,
-        message: "This is an insight notification",
+        message: "This is an insight notification with id = " + id,
         icon: "icon1",
         actions: [
           {
             actionText: "Open insight",
             actionType: Office.MailboxEnums.ActionType.ShowTaskPane,
-            commandId: "msgComposeOpenPaneButton",
+            commandId: "PG.HelpCommand.Compose",
```
Comment (kwame-ansong) [inline snippet-extractor-output/snippets.yaml:10489]: This seems to be the only difference between the two `details` objects. So, can we scope the `if...else` check to just these two, which will make it clearer?

File: snippet-extractor-output/snippets.yaml [hunk 3]
```diff
@@ -12691,20 +12715,44 @@
...
```
Comment (kwame-ansong) [inline snippet-extractor-output/snippets.yaml:12748]: Same comment here. Would it be easier and more readable to scope the `if...else` check to the `commandId` attribute?

File: samples/outlook/35-notifications/add-getall-remove.yaml [hunk 4]
```diff
@@ -55,20 +55,41 @@ script:
         function addInsight() {
           // Adds an informational message with actions to the mail item.
           const id = $("#notificationId").val().toString();
-          const details = 
-            {
+
+          let details;
+          // Identifies whether the current mail item is in read or compose mode.
```
Comment (ElizabethSamuel-MSFT) [inline samples/outlook/35-notifications/add-getall-remove.yaml:60]:
```suggestion
          // Identify whether the current mail item is in read or compose mode.
```

File: samples/outlook/35-notifications/add-getall-remove.yaml (hunk 1 above)
Comment (ElizabethSamuel-MSFT) [reply samples/outlook/35-notifications/add-getall-remove.yaml:151]:
```suggestion
          <h3>Try it out</h3>
```

Comment (ElizabethSamuel-MSFT) [reply samples/outlook/35-notifications/add-getall-remove.yaml:151]: OR...
```suggestion
          <h2>Try it out</h2>
```

File: samples/outlook/35-notifications/add-getall-remove.yaml [hunk 5]
```diff
@@ -127,34 +148,51 @@ template:
         </section>
 
         <section class="samples ms-font-m">
-          <h3>Try it out</h3>
+          <h2>Try it out</h3>
           <div class="ms-TextField">
-            <label class="ms-Label">Notification ID to use with add/replace/remove:</label>
+            <label class="ms-Label">Notification ID to add, replace, or remove a notification:</label>
```
Comment (ElizabethSamuel-MSFT) [inline samples/outlook/35-notifications/add-getall-remove.yaml:153]:
```suggestion
            <label class="ms-Label">Notification ID to add, replace, or remove that notification:</label>
```

File: samples/outlook/35-notifications/add-getall-remove.yaml [hunk 6]
```diff
@@ -55,20 +55,41 @@ script:
         function addInsight() {
           // Adds an informational message with actions to the mail item.
           const id = $("#notificationId").val().toString();
-          const details = 
-            {
+
+          let details;
+          // Identifies whether the current mail item is in read or compose mode.
+          if (Office.context.mailbox.item.itemId != undefined) {
+            // Read mode.
```
Comment (ElizabethSamuel-MSFT) [inline samples/outlook/35-notifications/add-getall-remove.yaml:62]: If you accept Kwame's suggestion re: scoping down the if/else, then you can do something like:
```suggestion
          let commandIdFromManifest = "PG.HelpCommand.Read";
          if (Office.context.mailbox.item.itemId == undefined) {
            commandIdFromManifest = "PG.HelpCommand.Compose";
          }
```

File: samples/outlook/35-notifications/add-getall-remove.yaml [hunk 7]
```diff
@@ -55,20 +55,41 @@ script:
...
               type: Office.MailboxEnums.ItemNotificationMessageType.InsightMessage,
-              message: "This is an insight notification",
+              message: "This is an insight notification with id = " + id,
               icon: "icon1",
               actions: [
                 {
                   actionText: "Open insight",
                   actionType: Office.MailboxEnums.ActionType.ShowTaskPane,
-                  commandId: "msgComposeOpenPaneButton",
+                  commandId: "PG.HelpCommand.Read",
```
Comment (ElizabethSamuel-MSFT) [inline samples/outlook/35-notifications/add-getall-remove.yaml:71]:
```suggestion
                  commandId: commandIdFromManifest,
```

File: samples/outlook/35-notifications/add-getall-remove.yaml [hunk 8]
```diff
@@ -55,20 +55,41 @@ script:
...
+              actions: [
+                {
+                  actionText: "Open insight",
+                  actionType: Office.MailboxEnums.ActionType.ShowTaskPane,
+                  commandId: "PG.HelpCommand.Compose",
+                  contextData: { a: "aValue", b: "bValue" }
+                }
+              ]
+            };
+          }
```
Comment (ElizabethSamuel-MSFT) [inline samples/outlook/35-notifications/add-getall-remove.yaml:91]: Again, if you accept Kwame's scoping suggestion, then no need for else
```suggestion
```

File: snippet-extractor-output/snippets.yaml (hunk 2 above)
Comment (samantharamon) [reply snippet-extractor-output/snippets.yaml]: I forgot that we can use the ternary operator within JS objects. Updated the code to use this for clarity.

File: samples/outlook/35-notifications/add-getall-remove.yaml [hunk 9]
```diff
@@ -127,34 +130,51 @@ template:
...
+          <br>
           <button id="addError" class="ms-Button">
             <div class="ms-Button-label">Add an error notification</div>
           </button>
+          <h3>Get all notifications</h3>
           <button id="getAll" class="ms-Button">
-            <div class="ms-Button-label">Get all</div>
+            <div class="ms-Button-label">Get notifications</div>
           </button>
+          <h3>Replace a notificationId</h3>
```
Comment (ElizabethSamuel-MSFT) [inline samples/outlook/35-notifications/add-getall-remove.yaml:169]:
```suggestion
          <h3>Replace a notification</h3>
```

**Review by ElizabethSamuel-MSFT — APPROVED** [review]

**Review by kwame-ansong — APPROVED** [review]

**Pull request merged**